### DISCUSSION TITLE
Traefik 2.10 Upgrade Path

### DISCUSSION THREAD

**Discussion by l2dy** [k3s-io/k3s#8500] [General]
Traefik 2.10 introduced a new CRD API group, `traefik.io`, and intends to deprecate the `traefik.containo.us` API group in Traefik 3.0.

Will an upgrade to Traefik >=2.10 in K3s packaged components be planned soon? How should we migrate the CRDs?

https://traefik.io/blog/announcing-traefik-proxy-2-10/

**Comment by brandond** [comment]
We haven't tackled that yet. There were similar issues around the Traefik 1 to 2 migration, and we mostly just punted on it by not upgrading to v2 if v1 was installed, so that users had to take manual steps to migrate their config and remove v1 if they wanted to use the new version. We may do something similar for this.

Honestly it sucks that they're switching the CRD API group on v2.10, this is kind of a breaking change and semver would suggest a new major version (3.x).

**Reply by l2dy** [reply]
Either `traefik.containo.us` or `traefik.io` can be used, so I think it's not a breaking change yet in 2.x.
